### DISCUSSION TITLE
Auto Persist Model Changes to Folder (Git Repo)

### DISCUSSION THREAD

**Discussion by tobimax** [TabularEditor/TabularEditor3#1053] [Ideas]
Hi Daniel,

At the moment the only way to persist to folder is to save to folder which sometimes dumps a load of stuff your not interested in.

When I open a new branch in a git repo to do some work it would be nice for TE3 to write the changes directly to the folder automatically. That way only the stuff I'm adding or changing will show up as items to stage and commit.

This could be enabled for both offline and connected models and would speed up the development process (GitHub Flow).

What do you think?

Regards

Toby

**Comment by otykier** [comment]
@tobimax I'm not following. Save-to-folder writes the entire model back to disk, but your git repo should automatically pick up files that were actually changed. I.e. if you modified the DAX expression of a measure, and you configured the save-to-folder serialization to create individual files for each measure, then you should only see a change on that one file representing the measure.

> sometimes dumps a load of stuff your not interested in

Could you provide an example of this?

> write the changes directly to the folder automatically

Do you mean write the changes as you make them (i.e. without having to hit Ctrl+S)? I don't like to add features in which developers "lose control" over what happens. A save is potentially a 2-5 second operation, depending on the size of the model, so I'm not sure I see the benefit of this.
